vnext/Microsoft.ReactNative.ComponentTests/CommonReaderTest.h

```diff
@@ -0,0 +1,324 @@
+    writer.WriteObjectEnd();
+  }
+
+  static void Read(IJSValueReader &reader) {
+    hstring propertyName;
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+    TestCheckEqual(false, reader.GetNextObjectProperty(propertyName));
+  }
+};
+
+// {a:null, b:true, c:12345}
+struct SimpleObject {
+  static void Write(IJSValueWriter &writer) {
+    writer.WriteObjectBegin();
+    writer.WritePropertyName(L"a");
+    writer.WriteNull();
+    writer.WritePropertyName(L"b");
+    writer.WriteBoolean(true);
+    writer.WritePropertyName(L"c");
+    writer.WriteInt64(12345);
+    writer.WriteObjectEnd();
+  }
+
+  static void Read(IJSValueReader &reader) {
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+    TestCheckObject(
+        {{L"a", [&]() { TestCheckEqual(JSValueType::Null, reader.ValueType()); }},
+         {L"b",
+          [&]() {
+            TestCheckEqual(JSValueType::Boolean, reader.ValueType());
+            TestCheckEqual(true, reader.GetBoolean());
+          }},
+         {L"c",
+          [&]() {
+            TestCheckEqual(JSValueType::Int64, reader.ValueType());
+            TestCheckEqual(12345, reader.GetInt64());
+          }}},
+        reader);
+  }
+};
+
+// {a:{x:1}, b:{x:2}, c:{x:3} ,d:{x:4}, e:{x:5}}
+struct NestedObjectWithPrimitiveValues {
+  static void Write(IJSValueWriter &writer) {
+    hstring props[] = {L"a", L"b", L"c", L"d", L"e"};
+    int index = 0;
+
+    writer.WriteObjectBegin();
+    for (auto prop : props) {
+      writer.WritePropertyName(prop);
+      writer.WriteObjectBegin();
+      writer.WritePropertyName(L"x");
+      writer.WriteInt64(++index);
+      writer.WriteObjectEnd();
+    }
+    writer.WriteObjectEnd();
+  }
+
+  static void Read(IJSValueReader &reader) {
+    hstring props[] = {L"a", L"b", L"c", L"d", L"e"};
+    int index = 0;
+
+    hstring propertyName;
+    std::map<hstring, std::function<void()>> assertions;
+    for (auto prop : props) {
+      ++index;
+      assertions.insert({prop, [&, index]() {
+                           TestCheckEqual(JSValueType::Object, reader.ValueType());
+                           TestCheckEqual(true, reader.GetNextObjectProperty(propertyName));
+                           TestCheckEqual(L"x", propertyName);
+                           TestCheckEqual(JSValueType::Int64, reader.ValueType());
+                           TestCheckEqual(index, reader.GetInt64());
+                           TestCheckEqual(false, reader.GetNextObjectProperty(propertyName));
+                         }});
+    }
+
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+    TestCheckObject(assertions, reader);
+  }
+};
+
+//
+
+} // namespace winrt::Microsoft::ReactNative::ReaderTestCases
+
+// IMPORT_READER_TEST_CASES macro runs the same set of test cases
+// on DynamicReader+DynamicWriter and JsiReader+JsiWriter
+// in this way we ensure that two implementation behaves exactly the same
+
+#define IMPORT_READER_TEST_CASE(NAME)       \
```

Comment:
>NAME [](start = 32, length = 4)

nit: we typically do not use all upper case letter for arguments.

Reply:
changed to `name`